src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,76 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+
+interface Props {
+  handleCloseModal: () => void;
+  selectedButton: number;
+  recentTodo: Todo;
+}
+export const TodoModal: FC<Props> = ({
+  handleCloseModal,
+  selectedButton,
+  recentTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const {
+    userId,
+    id,
+    title,
+    completed,
+  } = recentTodo || {};
```

Comment:
- do not try to access values of empty object
- here your first test fails – when `todos` array is empty, you actually pass nothing to `recentTodo`, and it cannot get value out of `undefined`
<img width="801" alt="Screenshot 2023-01-05 at 14 15 10" src="https://user-images.githubusercontent.com/75496546/210777986-c75dead5-a20d-4f3e-bb4a-a60c30c0979a.png">